### ISSUE TITLE
Rename StoredProcedure Does not Work

### ISSUE THREAD

**Issue by estyfen** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#569]
Hi,

There is an error in the Delegate that assigns this function, for that reason it does not take the custom name that is assigned in the template .tt. It should be similar to the StoredProcedureReturnModelRename function, which does work.

EF.Reverse.POCO.v3.ttinclude

        `public static Func<StoredProcedure, string> StoredProcedureRename = delegate (StoredProcedure sp)
        {
            // Example:
            //if (sp.NameHumanCase.StartsWith("sp_"))
            //    return sp.NameHumanCase.Remove(0, 3);
            //return sp.NameHumanCase.Replace("my_sp_", "");

            return sp.NameHumanCase; // Do nothing by default
        };`

**Comment by sjh37** [comment]
Thanks, I'll take a look

**Comment by estyfen** [comment]
> Thanks, I'll take a look

The problem is in function WriteStoredProcFunctionName() of Class  StoredProcedure

Class **StoredProcedure** 

Current code

```
        public string WriteStoredProcFunctionName()
        {
            return NameHumanCase;
        }
```

Improvement proposal
```
        public string WriteStoredProcFunctionName(IDbContextFilter filter)
        {
            var name = NameHumanCase;
            var customName = filter.StoredProcedureRename(this);
            if (!string.IsNullOrEmpty(customName))
                 name = customName;

            return name;
        }
```

**Comment by sjh37** [comment]
Fix is done. Thanks for your help @estyfen This will be in the next release.

**Comment by sjh37** [comment]
Released in v3.1.1
